### DISCUSSION TITLE
What's the usage of internal Currency type?

### DISCUSSION THREAD

**Discussion by huoyaoyuan** [dotnet/runtime#66285] [Q&A] [answered]
https://github.com/dotnet/runtime/blob/7698a9a80c5f6270aa1122d79ce419c7b03f2498/src/coreclr/System.Private.CoreLib/src/System/Currency.cs#L6-L16

I guess it's used by interop scenario, but there's no reference in managed code, and I can't find corresponding unmanaged code. The name hints that it's a part of OA interop, but it's not used by OAVariantLib either.

**Comment by deeprobin** [comment]
IIRC its some Visual Basic thing.

See https://stackoverflow.com/a/151314/7600021

**Reply by huoyaoyuan** [reply]
OK, It may be better to add comment about the VB6 usage.

**Comment by jkotas** [comment, marked as answer]
It is used when you use ` [MarshalAs(UnmanagedType.Currency)]` on Decimal. Look for `NATIVE_TYPE_CURRENCY`, `MARSHAL_TYPE_CURRENCY` and `ILCurrencyMarshaler` in the unmanaged runtime to see where it used.
